sdk/schemaregistry/azure-schemaregistry-avroserializer/azure/schemaregistry/serializer/avro_serializer/_schema_registry_avro_serializer.py

```diff
@@ -0,0 +1,154 @@
+# of this software and associated documentation files (the ""Software""), to
+# deal in the Software without restriction, including without limitation the
+# rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+# sell copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+# IN THE SOFTWARE.
+#
+# --------------------------------------------------------------------------
+from io import BytesIO
+from typing import Any, Dict, Union
+import avro
+
+from azure.schemaregistry import SchemaRegistryClient, SerializationType
+
+from ._avro_serializer import AvroObjectSerializer
+
+
+class SchemaRegistryAvroSerializer(object):
+    """
+    SchemaRegistryAvroSerializer provides the ability to serialize and deserialize data according
+    to the given avro schema. It would automatically register, get and cache the schema.
+
+    :param str endpoint: The Schema Registry service endpoint, for example my-namespace.servicebus.windows.net.
+    :param credential: To authenticate to manage the entities of the SchemaRegistry namespace.
+    :type credential: TokenCredential
+    :param str schema_group: Schema group under which schema should be registered.
+
+    """
+    def __init__(self, endpoint, credential, schema_group, **kwargs):
+        self._schema_group = schema_group
+        self._avro_serializer = AvroObjectSerializer()
+        self._schema_registry_client = SchemaRegistryClient(credential=credential, endpoint=endpoint, **kwargs)
+        self._id_to_schema = {}
+        self._schema_to_id = {}
+        self._user_input_schema_cache = {}
+
+    def __enter__(self):
+        # type: () -> SchemaRegistryAvroSerializer
+        self._schema_registry_client.__enter__()
+        return self
+
+    def __exit__(self, *exc_details):
+        # type: (Any) -> None
+        self._schema_registry_client.__exit__(*exc_details)
+
+    def close(self):
+        # type: () -> None
+        """ This method is to close the sockets opened by the client.
+        It need not be used when using with a context manager.
+        """
+        self._schema_registry_client.close()
+
+    def _get_schema_id(self, schema_name, schema, **kwargs):
+        # type: (str, avro.schema.Schema, Any) -> str
+        """
```

Comment:
I understand it's internal, but for the sake of maintenance even something quick would help with the behavior

Reply:
you're right, added some comment to explain